OsmAndMapCreator/src/net/osmand/data/diff/ObfDiffGenerator.java

```diff
@@ -188,10 +215,21 @@ private void compareMapData(ObfFileInMemory fStart, ObfFileInMemory fEnd, boolea
 					}
 				} else {
 					if (objE == null) {
-						// Object with this id is not present in the second obf
-						BinaryMapDataObject obj = new BinaryMapDataObject(idx, objS.getCoordinates(), null,
-								objS.getObjectType(), objS.isArea(), new int[] { deleteId }, null);
-						endData.put(idx, obj);
+						if (deletedObjIds != null) {
+							if (deletedObjIds.containsKey(osmid)) {
```

Comment:
Not needed check? why storing long & checking long if we need to check EntityId? WTF?